backend/controllers/addAppointmentDetails.js

```diff
@@ -0,0 +1,158 @@
+    const { activeCaseLoadId } = req.session.userDetails
+
+    try {
+      const {
+        appointmentType,
+        location,
+        date,
+        startTimeHours,
+        startTimeMinutes,
+        endTimeHours,
+        endTimeMinutes,
+        sameTimeAppointments,
+        comments,
+      } = req.body || {}
+
+      const errors = []
+      const now = moment()
+
+      const startTime = startTimeHours && startTimeMinutes && moment(date, DAY_MONTH_YEAR)
+      const endTime = endTimeHours && endTimeMinutes && moment(date, DAY_MONTH_YEAR)
+
+      if (startTime) {
+        startTime.hour(Number(startTimeHours))
+        startTime.minute(Number(startTimeMinutes))
+      }
+
+      if (endTime) {
+        endTime.hour(Number(endTimeHours))
+        endTime.minute(Number(endTimeMinutes))
+      }
+
+      if (!appointmentType) errors.push({ text: 'Select an appointment type', href: '#appointment-type' })
+      if (!location) errors.push({ text: 'Select a location', href: '#location' })
+      if (!startTime) errors.push({ text: 'Select a start time', href: '#start-time' })
+      if (!date) errors.push({ text: 'Select a date', href: '#date' })
+      if (!sameTimeAppointments)
+        errors.push({ text: 'Select yes if the appointments all have the same time', href: '#same-time-appointments' })
+
+      if (date && !moment(date, DAY_MONTH_YEAR).isValid())
+        errors.push({ text: 'Enter a date in DD/MM/YYYY format', href: '#date' })
+
+      if (date && moment(date, DAY_MONTH_YEAR).isBefore(now, 'day')) {
+        errors.push({ text: 'Select a date that is not in the past', href: '#date' })
+      } else {
+        if (startTime && startTime.isBefore(now, 'minutes'))
+          errors.push({ text: 'Select a start time that is not in the past', href: '#start-time' })
+
+        if (endTime && endTime.isBefore(startTime, 'minutes'))
+          errors.push({ text: 'Select a end time that is not in the past', href: '#end-time' })
+      }
+
+      if (comments && comments.length > 3600)
+        errors.push({ text: 'Maximum length should not exceed 3600 characters', href: '#comments' })
+
+      const { appointmentTypes, locations } = await getAppointmentTypesAndLocations(res.locals, activeCaseLoadId)
+
+      const appointmentTypeDetails = appointmentType && appointmentTypes.find(type => type.value === appointmentType)
+      const locationDetails = location && locations.find(type => type.value === Number(location))
+
+      if (!errors.length) {
+        // eslint-disable-next-line no-param-reassign
+        req.session.data = {
+          appointmentType,
+          appointmentTypeDescription: appointmentTypeDetails.text,
+          location,
+          locationDescription: locationDetails.text,
+          date,
+          startTime: startTime.format(DATE_TIME_FORMAT_SPEC),
+          endTime: endTime && endTime.format(DATE_TIME_FORMAT_SPEC),
+          sameTimeAppointments,
+        }
+
+        res.redirect('/upload-file')
```

Comment:
We need to come to consensus on the url.  John has prefixed with `bulk-appointments/` which makes sense but isn't mentioned in the ACs.  I think we should go with bulk-appointments/upload-file (etc).